### ISSUE TITLE
[MEAR-36] Add classifier fonctionnality to Maven EAR Plugin

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-ear-plugin#236]
**[Mathieu Rozieres](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mrozieres)** opened **[MEAR-36](https://issues.apache.org/jira/browse/MEAR-36?redirect=false)** and commented

The \<classifier> tag is not implemented into Maven EAR Plugin.

For example, while using this configuration :

\<plugin>
\<groupId>org.apache.maven.plugins\</groupId>
\<artifactId>maven-ear-plugin\</artifactId>
\<configuration>
\<classifier>dev\</classifier>
\</configuration>
\</plugin>
The artefact produced is still named ${pom.artifactId}-${pom.version} ...

---

**Affects:** 2.2

**Attachments:**
- [MEAR-36-maven-ear-plugin.patch](https://issues.apache.org/jira/secure/attachment/12715760/MEAR-36-maven-ear-plugin.patch) (_3.01 kB_)

2 votes, 1 watchers

**Comment by jira-importer** [comment]
**[Eric Bernstein](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ebernstein)** commented

Attaching a patch that fixes this issue.  
I know the issue says in progress but I needed the fix today and wrote it out.  The fix was taken from the maven-jar-plugin implementation and should function identically.

**Comment by jira-importer** [comment]
**[Eric Bernstein](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ebernstein)** commented

One note on the patch: Like the jar plugin, it expects to build ears with classifiers as attachments to the ear w/out the classifier.  I wasn't sure that's what was desired, but I figured there was a good reason the jar plugin did it as such, and it met my reqs, so I did the ear patch the same way.

**Comment by jira-importer** [comment]
**[Stephane Nicoll](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=snicoll)** commented

Eric, the issue is in progress because I started the implementation before going on holiday. It's implemented in my local sandbox, sorry for not menitionning it. I am not resolving it because I am busy writing test cases for the EAR plugin (namely this classifier fonctionnality) and I am stuck on some major points.

I will make sure that the changes does not break your patch if possible.

**Comment by jira-importer** [comment]
**[Eric Bernstein](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ebernstein)** commented

Sorrry Stephane, I didn't mean to step on any toes, but since I was writing it anyway so I could build my project, I figured I'd hand it back in case it could be of use.

Don't worry about keeping compatibility with what I submitted, I'm happy to update my poms when the next ear plugin is released - I hard coded the dependencies in my projects to my own internal version of the plugin for now, so I'll have to update eventually.

Just for my own curiosity, are you keeping classified ears as attachments or making it so that the classified ear could be the only artifact.  I was torn, and as mentioned, ended up just stealing what the jar plugin was doing.  Not sure which way I think is right.
